### ISSUE TITLE
MTK phones with MT6855 chipset do not support handtrackinggpu

### ISSUE THREAD

**Issue by z8888q** [google/mediapipe#4500]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

No

### OS Platform and Distribution

mediapipe v0.10.0 official Docker on Windows 11 WSL2 

### Mobile device if the issue happens on mobile device

Moto g73 5G, and other Android Phones with MTK MT6855 chipset, officially named "Dimensity 7020" "天玑7020"

### Programming Language and version

Java

### MediaPipe version

0.10.0

### Bazel version

6.1.1

### Solution

HandTracking, HandDetection

### Describe the actual behavior

```shell
Follow the official doc to build the handtrackinggpu app and install it to Moto g73 5G, grant the necessary permission, I put my hand in front of the phone screen. Now all I can see is a camera preview of the original hand, with no hand landmarks or frames of the hand on it.
```

### Describe the expected behaviour

```shell
There should be hand landmarks on the phone screen.
```

### Standalone code/steps you may have used to try to get what you need

```shell
1. Follow the installation doc from here: Installing using Docker (https://developers.google.com/mediapipe/framework/getting_started/install#installing_using_docker)

2. The build command is:
bazel build -c opt --config=android_arm64 mediapipe/examples/android/src/java/com/google/mediapipe/apps/handtrackinggpu:handtrackinggpu

3. After the build is completed, install the handtrackinggpu.apk on an Android device (Moto g73 5G)

4. open the handtrackinggpu app on the phone and grant the necessary permission, then check the behavior
```

### Other info / Complete Logs

```shell
I tested the app on 2 Android phones with the MT6855 chipset, both failed to show the hand landmarks.
I tested the app on Qcom and other MTK phones, they can show the hand landmarks.
```

**Comment by z8888q** [comment]
Hi @kuaashish 
After some more digging, I found out that the difference with a normal phone is::
https://github.com/google/mediapipe/blob/d3f98334bfe533a57ec2c2fa85294c21e25fe0af/mediapipe/calculators/tensor/tensors_to_detections_calculator.cc#L787
In the TensorsToDetectionsCalculator::ConvertToDetections() method,  all detection_scores are below 0.01 even when the hand is in front of the screen.
I have no idea about how to proceed to analyze this problem. What do you recommend?

Thank you！

**Comment by kuaashish** [comment]
Hello @jiuqiant,

Please look into this issue. Thank you!!

**Comment by img-candet** [comment]
Hi,

This is most likely caused by an issue in Imagination's libGLES, which ships with MT6855 devices. It sounds like the glDispatchCompute calls in TensorsToDetectionsCalculator::ProcessGPU aren't being correctly waited for by the driver.

We think this is happening because Mediapipe relies on glFenceSync/glClientWaitSync to wait for the compute shader to finish. Our driver doesn't correctly handle synchronisation in this case. This is a known issue that will be fixed in the next version (23.2).

As a workaround, I would suggest either calling glFinish right after the second glDispatchCompute call, or making sure the glMapBufferRange path is used in Tensor::GetCpuReadView. If neither of those help, I would try calling glFinish after every glDispatchCompute call the library does, to narrow down which ones are affected by the issue.

**Comment by z8888q** [comment]
MTK patch verified to work.
I think the issue can be closed.
